lib/osf-components/addon/components/validated-input/checkboxes/x-checkbox/component.ts

```diff
@@ -10,26 +11,37 @@ import template from './template';
 @tagName('')
 export default class ValidatedInputCheckboxesXCheckbox<T> extends Component {
     // Required arguments
-    relationArray?: T[];
+    relationArray?: DS.PromiseArray<T> | T[];
     option!: T;
     checkboxName!: string;
     ariaLabel!: string;
     disabled!: boolean;
     checkboxId!: string;
 
-    @computed('option', 'relationArray.[]')
+    @computed('option', '_relationArray.[]')
     get checked(): boolean {
-        return Array.isArray(this.relationArray) && this.relationArray.includes(this.option);
+        return Array.isArray(this._relationArray) && this._relationArray.includes(this.option);
```

Comment:
```suggestion
        return this._relationArray.includes(this.option);
```
the `isArray` check is now redundant